### PR TITLE
Neilj/mau tracking config explainer

### PR DESCRIPTION
**Pull request by neilisfragile**

### CODE REVIEW

File: synapse/config/server.py [hunk 1]
```diff
@@ -584,6 +584,13 @@ def default_config(self, server_name, data_dir_path, **kwargs):
         #hs_disabled_limit_type: 'error code(str), to help clients decode reason'
 
         # Monthly Active User Blocking
+        # Used in cases where the admin or server owner wants to limit to the
```
Comment (richvdh) [inline synapse/config/server.py]:
```suggestion
        #
        # Used in cases where the admin or server owner wants to limit to the
```

File: synapse/config/server.py [hunk 2]
```diff
@@ -584,6 +584,13 @@ def default_config(self, server_name, data_dir_path, **kwargs):
...
+        # number of monthly active users.
+        # 'mau_trial_days' is means to add a grace period for active users. It
```
Comment (richvdh) [inline synapse/config/server.py]:
```suggestion
        #
        # 'mau_trial_days' is means to add a grace period for active users. It
```

File: synapse/config/server.py [hunk 3]
```diff
@@ -584,6 +584,13 @@ def default_config(self, server_name, data_dir_path, **kwargs):
         #hs_disabled_limit_type: 'error code(str), to help clients decode reason'
 
         # Monthly Active User Blocking
+        # Used in cases where the admin or server owner wants to limit to the
+        # number of monthly active users.
```
Comment (richvdh) [inline synapse/config/server.py]: Could you document what happens when the limit is hit?

File: docs/sample_config.yaml [hunk 4]
```diff
@@ -261,10 +261,26 @@ listeners:
 
 # Monthly Active User Blocking
 #
+# Used in cases where the admin or server owner wants to limit to the
```
Comment (anoadragon453) [inline docs/sample_config.yaml]:
```suggestion
# Used in cases where the admin or server owner wants to limit the
```

File: docs/sample_config.yaml [hunk 5]
```diff
@@ -261,10 +261,26 @@ listeners:
...
+# number of monthly active users.
+#
+# 'limit_usage_by_mau' disables/enables monthly active user blocking. When
+# anabled and a limit is reached the server returns a 'ResourceLimitError'
```
Comment (anoadragon453) [inline docs/sample_config.yaml]:
```suggestion
# enabled and a limit is reached the server returns a 'ResourceLimitError'
```

File: docs/sample_config.yaml [hunk 6]
```diff
@@ -261,10 +261,26 @@ listeners:
...
+# with error type Codes.RESOURCE_LIMIT_EXCEEDED
```
Comment (anoadragon453) [inline docs/sample_config.yaml]:
```suggestion
# with error type Codes.RESOURCE_LIMIT_EXCEEDED.
```

File: docs/sample_config.yaml [hunk 7]
```diff
@@ -261,10 +261,26 @@ listeners:
...
+#
+# 'max_mau_value' is the hard limit of monthly active users above which
+# the server will start blocking user actions.
+#
+# 'mau_trial_days' is a means to add a grace period for active users. It
```
Comment (anoadragon453) [inline docs/sample_config.yaml]:
```suggestion
# 'mau_trial_days' is a means for adding a grace period for active users. It
```

File: synapse/config/server.py [hunk 8]
```diff
@@ -585,10 +585,26 @@ def default_config(self, server_name, data_dir_path, **kwargs):
 
         # Monthly Active User Blocking
         #
+        # Used in cases where the admin or server owner wants to limit to the
+        # number of monthly active users.
+        #
+        # 'limit_usage_by_mau' disables/enables monthly active user blocking. When
+        # anabled and a limit is reached the server returns a 'ResourceLimitError'
```
Comment (anoadragon453) [inline synapse/config/server.py]: as above

File: synapse/config/server.py [hunk 9]
```diff
@@ -585,10 +585,26 @@ def default_config(self, server_name, data_dir_path, **kwargs):
 
         # Monthly Active User Blocking
         #
+        # Used in cases where the admin or server owner wants to limit to the
```
Comment (anoadragon453) [inline synapse/config/server.py]: as above

File: synapse/config/server.py [hunk 10]
```diff
@@ -585,10 +585,26 @@ def default_config(self, server_name, data_dir_path, **kwargs):
...
+        # number of monthly active users.
+        #
+        # 'limit_usage_by_mau' disables/enables monthly active user blocking. When
+        # anabled and a limit is reached the server returns a 'ResourceLimitError'
+        # with error type Codes.RESOURCE_LIMIT_EXCEEDED
```
Comment (anoadragon453) [inline synapse/config/server.py]: as above

File: synapse/config/server.py [hunk 11]
```diff
@@ -585,10 +585,26 @@ def default_config(self, server_name, data_dir_path, **kwargs):
...
+        #
+        # 'max_mau_value' is the hard limit of monthly active users above which
+        # the server will start blocking user actions.
+        #
+        # 'mau_trial_days' is a means to add a grace period for active users. It
```
Comment (anoadragon453) [inline synapse/config/server.py]: as above

File: synapse/config/server.py [hunk 12]
```diff
@@ -585,10 +585,26 @@ def default_config(self, server_name, data_dir_path, **kwargs):
...
+        # means that users must be active for this number of days before they
+        # can be considered active and guards against the case where lots of users
+        # sign up in a short space of time never to return after their initial
+        # session.
+        #
         #limit_usage_by_mau: False
         #max_mau_value: 50
         #mau_trial_days: 2
-
+        #
```
Comment (anoadragon453) [inline synapse/config/server.py]: Remove this

**Pull request merged**